contracts/monitoring/src/error.rs

```diff
@@ -0,0 +1,15 @@
+use axelar_wasm_std_derive::IntoContractError;
+use cosmwasm_std::StdError;
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, IntoContractError)]
+pub enum ContractError {
+    #[error(transparent)]
+    Std(#[from] StdError),
+
+    #[error("unauthorized")]
+    Unauthorized,
+
+    #[error("no provers registered for chain")]
+    NoProversRegisteredForChain,
```

Comment:
```suggestion
    #[error("no provers registered for chain {0}")]
    NoProversRegisteredForChain(ChainName),
```
not sure if this works this way or if it needs to be explicitly transformed into a string. Either way, adding the chain name helps making the error clearer